datalad/distributed/create_sibling_github.py

```diff
@@ -0,0 +1,266 @@
+                 > ds.create_sibling_gin('myrepo', name='gin')
+
+                 # the sibling on GitHub will be used for collaborative work
+                 > ds.create_sibling_github('myrepo', name='github')
+
+                 # register the storage of the public GIN repo as a data source
+                 > ds.siblings('configure', name='gin', as_common_datasrc='gin-storage')
+
+                 # announce its availability on github
+                 > ds.push(to='github')
+                 """,
+             code_cmd="""\
+                 % datalad create-sibling-gin myrepo -s gin
+
+                 # the sibling on GitHub will be used for collaborative work
+                 % datalad create-sibling-github myrepo -s github
+
+                 # register the storage of the public GIN repo as a data source
+                 % datalad siblings configure -s gin --as-common-datasrc gin-storage
+
+                 # announce its availability on github
+                 % datalad push --to github
+                 """,
+             ),
+    ]
+
+    _params_ = _GitHub.create_sibling_params
+    _params_['api']._doc = """\
+        URL of the GitHub instance API"""
+    # special casing for deprecated mode
+    _params_['existing'].constraints = EnsureChoice(
+        'skip', 'error', 'reconfigure', 'replace')
+    _params_['existing']._doc += """\
+        DEPRECATED DANGER ZONE: With 'replace', an existing repository will be
+        irreversibly removed, re-initialized, and the sibling
+        (re-)configured (thus implies 'reconfigure').
+        `replace` could lead to data loss! In interactive sessions a
+        confirmation prompt is shown, an exception is raised in non-interactive
+        sessions. The 'replace' mode will be removed in a future release."""
+    # deprecated options
+    _params_.update(
+        github_login=Parameter(
+            args=('--github-login',),
+            constraints=EnsureStr() | EnsureNone(),
+            metavar='TOKEN',
+            doc="""Deprecated, use the credential parameter instead.
+            If given must be a personal access token."""),
+        github_organization=Parameter(
+            args=('--github-organization',),
+            constraints=EnsureStr() | EnsureNone(),
+            metavar='NAME',
+            doc="""Deprecated, prepend a repo name with an '<orgname>/'
+            prefix instead."""),
+        dryrun=Parameter(
+            args=("--dryrun",),
+            action="store_true",
+            doc="""Deprecated. Use the renamed
+            [CMD: --dry-run CMD][PY: `dry_run` PY] parameter"""),
+    )
+
+    @staticmethod
+    @datasetmethod(name='create_sibling_github')
+    @eval_results
+    def __call__(
+            reponame,
+            dataset=None,
+            recursive=False,
+            recursion_limit=None,
+            name='github',
+            existing='error',
+            github_login=None,
+            credential=None,
```

Comment:
Why is the credential name "None" here? In case of 'gin' and 'gogs' it has sensible default names. I just copy-pasted a wrong token, and it took me a while to find the credential under the name "datalad-None" to delete it.

Reply:
I would not mind making it `'github'`. However, the previous implementation had so many ways to pull credentials from, I had no good concept. Probably safe.

Reply:
I decided to implement a uniform policy of using the hostname part of the API url as the default credential name. So `None` cannot occur anymore.